src/views/RecIM/components/List/index.js

```diff
@@ -0,0 +1,24 @@
+import { List } from '@material-ui/core';
+import { ActivityListing, MatchListing, ParticipantListing, TeamListing } from './Listing';
+
+const ActivityList = ({ activities }) => {
+  let content = activities.map((activity) => <ActivityListing activity={activity} />);
+  return <List>{content}</List>;
+};
+
+const ParticipantList = ({ participants }) => {
+  let content = participants.map((participant) => <ParticipantListing participant={participant} />);
+  return <List>{content}</List>;
+};
+
+const MatchList = ({ matches }) => {
+  let content = matches.map((match) => <MatchListing match={match} />);
+  return <List>{content}</List>;
+};
+
+const TeamList = ({ teams }) => {
+  let content = teams.map((team) => <TeamListing team={team} />);
+  return <List>{content}</List>;
+};
+
+export { ActivityList, ParticipantList, MatchList, TeamList };
```

Comment:
I want to say you could make this cleaner by making a generalized List component for us or maybe make our Listing component more generalized/simplified. 

How you could do this would be via:

_in the Listing component_
Changing the ______Listing to Listing = ({data, type}) 
^data would be the full array of objects
using a switch case on type to determine if it is a Participant,Match,Activity...so on and so forth

Then you could probably just do the mapping within either an overhead function in Listing or somewhere else.

Reply:
I think this would almost be the same amount of code with less clarity. Abstracting like this also gives us flexibility if a component needs react state or other hooks.